aten/src/ATen/native/cuda/UnaryGeometricKernels.cu

```diff
@@ -165,6 +166,67 @@ void tanh_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+// We manually overload arccosh because std::acosh does not work with thrust::complex types.
+template<typename scalar_t>
+__host__ __device__ static inline scalar_t arccosh_wrapper(scalar_t v) {
+  return ::acosh(v);
+}
+
+template<typename T>
+__host__ __device__ static inline thrust::complex<T> arccosh_wrapper(thrust::complex<T> v) {
+  return thrust::acosh(v);
+}
+
+void arccosh_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(ScalarType::Half, iter.dtype(), "arccosh_cuda", [&]() {
+    using thrust_t = typename ztype_cuda<scalar_t>::thrust_t;
+    gpu_kernel(iter, []GPU_LAMBDA(thrust_t a) -> thrust_t {
+      return arccosh_wrapper(a);
+    });
+  });
+}
+
+// We manually overload arcsinh because std::asinh does not work with thrust::complex types.
+template<typename scalar_t>
+__host__ __device__ static inline scalar_t arcsinh_wrapper(scalar_t v) {
+  return ::asinh(v);
+}
+
+template<typename T>
+__host__ __device__ static inline thrust::complex<T> arcsinh_wrapper(thrust::complex<T> v) {
+  return thrust::asinh(v);
+}
+
+void arcsinh_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(ScalarType::Half, iter.dtype(), "arcsinh_cuda", [&]() {
+    using thrust_t = typename ztype_cuda<scalar_t>::thrust_t;
+    gpu_kernel(iter, []GPU_LAMBDA(thrust_t a) -> thrust_t {
+      return arcsinh_wrapper(a);
+    });
+  });
+}
+
+template<typename scalar_t>
```

Comment:
Is this missing a comment over overloading here like there is for asinh and acosh?

Reply:
You are right, sorry I missed the comment here. Thanks for pointing it out. Will fix in the next commit. :+1:

Reply:
Should be resolved here: https://github.com/pytorch/pytorch/pull/38388/commits/7df086c1a36fdef1f3b7badb76fa1812b83a14f7